### PR TITLE
[DYNAREC] Fixed CF flag computation

### PR DESCRIPTION
**Pull request by ksco**
A very subtle error, the source register does not guarantee that the high bits are clean. Found while trying to run the program in #1450 on RV64.

### CODE REVIEW

File: src/dynarec/rv64/dynarec_rv64_emit_math.c [hunk 1]
```diff
@@ -45,7 +45,9 @@ void emit_add32(dynarec_rv64_t* dyn, int ninst, rex_t rex, int s1, int s2, int s
             BEQZ(s5, 8);
             ORI(xFlags, xFlags, 1 << F_CF);
         } else {
-            ADD(s5, s1, s2);
+            AND(s3, s1, xMASK);
+            AND(s4, s2, xMASK);
+            ADD(s5, s3, s4);
```
Comment (ptitSeb) [inline src/dynarec/rv64/dynarec_rv64_emit_math.c:50]: only 1 and with xMASK is needed actually...

Comment (ksco) [reply src/dynarec/rv64/dynarec_rv64_emit_math.c:50]: hmm why?

Comment (ptitSeb) [reply src/dynarec/rv64/dynarec_rv64_emit_math.c:50]: s3 will have upper bit to 0. anything and that will still have the upper bit to 0.

Comment (ksco) [reply src/dynarec/rv64/dynarec_rv64_emit_math.c:50]: oh I think you misread line 50, it’s add, not and.

Comment (ptitSeb) [reply src/dynarec/rv64/dynarec_rv64_emit_math.c:50]: ah right, add, not and :(

**Pull request merged**